onnxruntime/core/common/profiler.h

```diff
@@ -62,7 +62,7 @@ class Profiler {
   /*
   Return the stored start time of profiler
   */
-  uint64_t GetStartTime() const {
+  uint64_t GetStartTimeNs() const {
```

Comment:
> GetStartTimeNs [](start = 10, length = 15)

I do not think this is true. On Windows the timer will not get you that resolution. This is the reason we are using TIME_SPEC API for measuring precisely.

Reply:
Oh you are right. That's why I always get XXXXX00 for the profiler's time on my Windows. Even though the time is not as precise as nano second on some platforms, I suppose the function's name still need to include 'Ns' to indicate the unit of time (Linux can get that resolution).  I just added a comment to remind users.

Still, it is a thing we need to keep in mind and maybe use TIME_SPEC for the profiler's time in a new PR. Thank you for the information!

Reply:
Or do you think we should use milliseconds instead?

Reply:
The problem is that the API is alerady names as Ns which it does not give. I think the right thing to do is keep the name and reimplement it with TIME_SPEC.

---
In reply to: [497068476](https://github.com/microsoft/onnxruntime/pull/5280#discussion_r497068476) [](ancestors = 497068476)

Reply:
@yuslepukhin Actually the time (`Env::Default().NowMicros()`) TIME_SPEC uses is microsecond. The original time (`std::chrono::high_resolution_clock::now()`) Profiler's is using can be as precise as microsecond as well. However, neither can get the resolution of nanoseconds.

Therefore I tend to still use the original `std::chrono::high_resolution_clock::now()` to calculate the profiler's start time and rename the function from `_Ns` to `MicroS`. In that case, we can get the same resolution as microseconds and won't influence the return type and modify more code, which might break some existing usage.

If you have other concern, please comment. Thank you!